p2p/proto/receipt.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+
+message MessageToL1 {
+  Felt252         from_address    = 1;
+  repeated        Felt252 payload = 2;
+  EthereumAddress to_address      = 3;
+}
+
+message EthereumAddress {
+  bytes elements = 1;
+}
+
+message MessageToL2 {
+  EthereumAddress from_address         = 1;
+  repeated        Felt252 payload      = 2;
+  Felt252         to_address           = 3;
+  Felt252         entry_point_selector = 4;
+  Felt252         nonce                = 5;
+}
+
+message Receipt {
+  message ExecutionResources {
+    message BuiltinCounter {
+      uint32 bitwise     = 1;
+      uint32 ecdsa       = 2;
+      uint32 ec_op       = 3;
+      uint32 pedersen    = 4;
+      uint32 range_check = 5;
+      uint32 poseidon    = 6;
+      uint32 keccak      = 7;
+    }
+
+    BuiltinCounter builtins     = 1;
+    uint32         steps        = 2;
+    uint32         memory_holes = 3;
+  }
+
+  message Common {
+    Hash               transaction_hash          = 1;
+    Felt252            actual_fee                = 2;
+    repeated           MessageToL1 messages_sent = 3;
+    ExecutionResources execution_resources       = 4;
+    string             revert_reason             = 5;
+  }
+
+
+  message Invoke {
+    Common common = 1;
+  }
+
+  message L1Handler {
+    Common common   = 1;
+    Hash   msg_hash = 2;
+  }
+
+  message Declare {
+    Common common = 1;
+  }
+
+  message Deploy {
+    Common common = 1;
+    Felt252 contract_address = 2;
+  }
+
+  message DeployAccount {
+    Common  common           = 1;
+    Felt252 contract_address = 2;
+  }
+
+  oneof receipt {
```

Comment:
Let us rename this because it is overloaded with parent messages. This makes it confusing and difficult to understand the generated structs in Go.
```suggestion
  oneof Type {
```